File: openapi-client/php-dt/src/App/DTO/MailLogEntry.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * A single email record in the mail log.  Combines data from the message store (envelope metadata), the queue release table (delivery status and response), and the sender delivery table (MX routing details).  Key field relationships with other API calls: - The &#x60;id&#x60; field matches the &#x60;mailid&#x60; query parameter on &#x60;GET /mail/log&#x60; and   the &#x60;text&#x60; field of a successful send response. - The &#x60;from&#x60; address can be passed to &#x60;POST /mail/blocks/delete&#x60; to delist a   flagged sender. - The &#x60;user&#x60; field is the SMTP username (e.g. &#x60;mb5658&#x60;) corresponding to the   &#x60;username&#x60; field in &#x60;GET /mail&#x60; / &#x60;GET /mail/{id}&#x60;.
 */
class MailLogEntry
{
    /**
     * Internal auto-increment database row ID.  Not meaningful outside the API.
     */
    #[DTA\Data(field: "_id")]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $_id = null;

    /**
     * The relay-assigned mail ID (18–19 hex characters).  This is the value returned as &#x60;text&#x60; by the sending endpoints and accepted as the &#x60;mailid&#x60; filter on &#x60;GET /mail/log&#x60;.
     */
    #[DTA\Data(field: "id")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $id = null;

    /**
     * SMTP envelope &#x60;MAIL FROM&#x60; address (may differ from the &#x60;From:&#x60; header).
     */
    #[DTA\Data(field: "from")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $from = null;

    /**
     * SMTP envelope &#x60;RCPT TO&#x60; address.
     */
    #[DTA\Data(field: "to")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $to = null;

    /**
     * Human-readable creation timestamp in &#x60;YYYY-MM-DD HH:MM:SS&#x60; format.
     */
    #[DTA\Data(field: "created")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $created = null;

    /**
     * Unix timestamp of message acceptance.  Corresponds to the &#x60;startDate&#x60; and &#x60;endDate&#x60; filter parameters on &#x60;GET /mail/log&#x60;.
     */
    #[DTA\Data(field: "time")]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $time = null;

    /**
     * The SMTP AUTH username used to submit the message (e.g. &#x60;mb5658&#x60;). Corresponds to the &#x60;username&#x60; field in &#x60;GET /mail&#x60; orders.
     */
    #[DTA\Data(field: "user")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $user = null;

    /**
     * SMTP transaction type negotiated with the relay (e.g. &#x60;ESMTPSA&#x60;).
     */
    #[DTA\Data(field: "transtype")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $transtype = null;

    /**
     * IP address of the client that submitted the message to the relay. Corresponds to the &#x60;origin&#x60; filter parameter on &#x60;GET /mail/log&#x60;.
     */
    #[DTA\Data(field: "origin")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $origin = null;

    /**
     * Relay interface name that accepted the message (e.g. &#x60;feeder&#x60;).
     */
    #[DTA\Data(field: "interface")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $interface = null;

    /**
     * The &#x60;Subject&#x60; header value, if available.
     */
    #[DTA\Data(field: "subject", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $subject = null;

    /**
     * The &#x60;Message-ID&#x60; header value, if present.  Can be used with the &#x60;messageId&#x60; filter on &#x60;GET /mail/log&#x60; for subsequent lookups.
     */
    #[DTA\Data(field: "messageId", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $message_id = null;

    /**
     * The sending zone assigned by the relay for outbound delivery.
     */
    #[DTA\Data(field: "sendingZone", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $sending_zone = null;

    /**
     * Size of the message body in bytes.
     */
    #[DTA\Data(field: "bodySize", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $body_size = null;

    /**
     * Sequence index of this recipient in a multi-recipient message. Starts at 1.
     */
    #[DTA\Data(field: "seq", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $seq = null;

    /**
     * Delivery status flag.  &#x60;1&#x60; &#x3D; successfully delivered to destination MX. &#x60;0&#x60; &#x3D; queued, deferred, or failed.  &#x60;null&#x60; &#x3D; delivery not yet attempted. Corresponds to the &#x60;delivered&#x60; filter parameter on &#x60;GET /mail/log&#x60;.
     */
    #[DTA\Data(field: "delivered", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $delivered = null;

    /**
     * The SMTP response string received from the destination MX server upon delivery attempt (e.g. &#x60;\&quot;250 2.0.0 Ok queued as C91D83E128C\&quot;&#x60;).
     */
    #[DTA\Data(field: "response", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $response = null;

    /**
     * The specific recipient address this delivery record is for.
     */
    #[DTA\Data(field: "recipient", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $recipient = null;

    /**
     * The destination domain.  Corresponds to the &#x60;mx&#x60; filter parameter (which matches &#x60;mxHostname&#x60;, not &#x60;domain&#x60;) on &#x60;GET /mail/log&#x60;.
     */
    #[DTA\Data(field: "domain", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $domain = null;

    /**
     * Whether the queue entry is currently locked for delivery processing.
     */
    #[DTA\Data(field: "locked", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $locked = null;

    /**
     * Millisecond-precision timestamp of the last queue lock acquisition.
     */
    #[DTA\Data(field: "lockTime", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $lock_time = null;

    /**
     * The relay server node assigned to deliver this message.
     */
    #[DTA\Data(field: "assigned", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $assigned = null;

    /**
     * ISO 8601 timestamp when the message was placed into the delivery queue.
     */
    #[DTA\Data(field: "queued", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $queued = null;

    /**
     * The MX hostname the relay connected to for delivery.  Corresponds to the &#x60;mx&#x60; filter parameter on &#x60;GET /mail/log&#x60;.
     */
    #[DTA\Data(field: "mxHostname", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $mx_hostname = null;

}

File: openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/MailLogEntry.scala

package org.openapitools.client.model


case class MailLogEntry (
    /* Internal auto-increment database row ID.  Not meaningful outside the API. */
    __id: Integer,
    /* The relay-assigned mail ID (18–19 hex characters).  This is the value returned as `text` by the sending endpoints and accepted as the `mailid` filter on `GET /mail/log`. */
    _id: String,
    /* SMTP envelope `MAIL FROM` address (may differ from the `From:` header). */
    _from: String,
    /* SMTP envelope `RCPT TO` address. */
    _to: String,
    /* Human-readable creation timestamp in `YYYY-MM-DD HH:MM:SS` format. */
    _created: String,
    /* Unix timestamp of message acceptance.  Corresponds to the `startDate` and `endDate` filter parameters on `GET /mail/log`. */
    _time: Integer,
    /* The SMTP AUTH username used to submit the message (e.g. `mb5658`). Corresponds to the `username` field in `GET /mail` orders. */
    _user: String,
    /* SMTP transaction type negotiated with the relay (e.g. `ESMTPSA`). */
    _transtype: String,
    /* IP address of the client that submitted the message to the relay. Corresponds to the `origin` filter parameter on `GET /mail/log`. */
    _origin: String,
    /* Relay interface name that accepted the message (e.g. `feeder`). */
    _interface: String,
    /* The `Subject` header value, if available. */
    _subject: Option[String],
    /* The `Message-ID` header value, if present.  Can be used with the `messageId` filter on `GET /mail/log` for subsequent lookups. */
    _messageId: Option[String],
    /* The sending zone assigned by the relay for outbound delivery. */
    _sendingZone: Option[String],
    /* Size of the message body in bytes. */
    _bodySize: Option[Integer],
    /* Sequence index of this recipient in a multi-recipient message. Starts at 1. */
    _seq: Option[Integer],
    /* Delivery status flag.  `1` = successfully delivered to destination MX. `0` = queued, deferred, or failed.  `null` = delivery not yet attempted. Corresponds to the `delivered` filter parameter on `GET /mail/log`. */
    _delivered: Option[Integer],
    /* The SMTP response string received from the destination MX server upon delivery attempt (e.g. `\"250 2.0.0 Ok queued as C91D83E128C\"`). */
    _response: Option[String],
    /* The specific recipient address this delivery record is for. */
    _recipient: Option[String],
    /* The destination domain.  Corresponds to the `mx` filter parameter (which matches `mxHostname`, not `domain`) on `GET /mail/log`. */
    _domain: Option[String],
    /* Whether the queue entry is currently locked for delivery processing. */
    _locked: Option[Integer],
    /* Millisecond-precision timestamp of the last queue lock acquisition. */
    _lockTime: Option[String],
    /* The relay server node assigned to deliver this message. */
    _assigned: Option[String],
    /* ISO 8601 timestamp when the message was placed into the delivery queue. */
    _queued: Option[String],
    /* The MX hostname the relay connected to for delivery.  Corresponds to the `mx` filter parameter on `GET /mail/log`. */
    _mxHostname: Option[String]
)
object MailLogEntry {
    def toStringBody(var__id: Object, var_id: Object, var_from: Object, var_to: Object, var_created: Object, var_time: Object, var_user: Object, var_transtype: Object, var_origin: Object, var_interface: Object, var_subject: Object, var_messageId: Object, var_sendingZone: Object, var_bodySize: Object, var_seq: Object, var_delivered: Object, var_response: Object, var_recipient: Object, var_domain: Object, var_locked: Object, var_lockTime: Object, var_assigned: Object, var_queued: Object, var_mxHostname: Object) =
        s"""
        | {
        | "_id":$var__id,"id":$var_id,"from":$var_from,"to":$var_to,"created":$var_created,"time":$var_time,"user":$var_user,"transtype":$var_transtype,"origin":$var_origin,"interface":$var_interface,"subject":$var_subject,"messageId":$var_messageId,"sendingZone":$var_sendingZone,"bodySize":$var_bodySize,"seq":$var_seq,"delivered":$var_delivered,"response":$var_response,"recipient":$var_recipient,"domain":$var_domain,"locked":$var_locked,"lockTime":$var_lockTime,"assigned":$var_assigned,"queued":$var_queued,"mxHostname":$var_mxHostname
        | }
        """.stripMargin
}
